### PR TITLE
LPS-132326 Use UI login for MFA test cases

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
https://issues.liferay.com/browse/LPS-132326

### CODE REVIEW

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:security

**Comment by gaborlovasdotcom** [comment]
ci:test:security:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by JoyceWang08211** [comment]
Start reviewing.

File: portal-web/test/functional/com/liferay/portalweb/macros/MultiFactorAuthentication.macro [hunk 1]
```diff
@@ -304,6 +305,14 @@ definition {
 
 			AssertElementNotPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_VERIFY_FIELD");
 
+			if (IsElementPresent(locator1 = "Button#I_AGREE")) {
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/macros/MultiFactorAuthentication.macro:308]: The new added user is required password reset, prefer to use JSONUser.setFirstPassword after adding new user. Then no need to add Line308-315.

File: portal-web/test/functional/com/liferay/portalweb/macros/User.macro [hunk 2]
```diff
@@ -1418,6 +1418,40 @@ definition {
 			userScreenName = "${userScreenName}");
 	}
 
+	macro editPasswordCPWithoutLogin {
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/macros/User.macro:1421]: Replace by using JSONUser.setFirstPassword

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/mfa/MultiFactorAuthentication.testcase [hunk 3]
```diff
@@ -857,12 +867,6 @@ definition {
 			userLastName = "userln",
 			userScreenName = "usersn");
 
-		User.openUsersAdmin();
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/mfa/MultiFactorAuthentication.testcase:860]: Replace by JSONUser.setFirstPassword

**Comment by JoyceWang08211** [comment]
Suggestion:
1. Use 
```
           JSONUser.setFirstPassword(
			agreeToTermsAndAnswerReminderQuery = "true",
			requireReset = "false",
			userEmailAddress = "userea@liferay.com");
```
instead of using
```
                User.openUsersAdmin();

		User.editPasswordCP(
			userEmailAddress = "userea@liferay.com",
			userScreenName = "usersn");
```
Then we can login with password "test"
```
               User.firstLoginUI(
			password = "test",
			userEmailAddress = "userea@liferay.com");
```
2. Use.editPasswordCPWithoutLogin can be replaced by using JSONUser.setFirstPassword.
3. No need to add if condition in line308-315 underMultiFactorAuthentication.macro when using JSONUser.setFirstPassword.

File: portal-web/test/functional/com/liferay/portalweb/macros/User.macro (hunk 2 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/macros/User.macro:1421]: I remove this method

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/mfa/MultiFactorAuthentication.testcase (hunk 3 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/mfa/MultiFactorAuthentication.testcase:860]: Replaced

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by JoyceWang08211** [comment]
After investigating the failure of [LocalFile.MultiFactorAuthentication#EndUserPermissionCheck](https://testray.liferay.com/home/-/testray/case_results/1534030446), we need to set the following property when using **JSONUser.setFirstPassword**:
`property custom.properties = "jsonws.web.service.paths.excludes=";`

**Comment by gaborlovasdotcom** [comment]
ci:test:mfa:nocompile

**Comment by gaborlovasdotcom** [comment]
Continued in https://github.com/JoyceWang08211/liferay-portal/pull/285

**Pull request closed without merging**